pdns/dnsscope.cc

```diff
@@ -257,85 +278,91 @@ try
 	  lowestTime=min((time_t)lowestTime,  (time_t)pr.d_pheader.ts.tv_sec);
 	  highestTime=max((time_t)highestTime, (time_t)pr.d_pheader.ts.tv_sec);
 
-	  string name=mdp.d_qname.toString()+"|"+DNSRecordContent::NumberToType(mdp.d_qtype);
-        
-	  QuestionIdentifier qi=QuestionIdentifier::create(pr.getSource(), pr.getDest(), mdp);
+	  QuestionIdentifier qi=QuestionIdentifier::create(pr.getSource(), pr.getDest(), header, qname, qtype);
 
-	  if(!mdp.d_header.qr) { // question
-	    if(!mdp.d_header.rd)
+	  if(!header.qr) { // question
+	    //	    cout<<"Query "<<qi<<endl;
+	    if(!header.rd)
 	      nonRDQueries++;
 	    queries++;
 
 	    ComboAddress rem = pr.getSource();
 	    rem.sin4.sin_port=0;
 	    requestors.insert(rem);	  
 
-	    QuestionData& qd=statmap[qi];
+            QuestionData& qd=statmap[qi];
           
 	    if(!qd.d_firstquestiontime.tv_sec)
 	      qd.d_firstquestiontime=pr.d_pheader.ts;
-	    qd.d_qcount++;
+	    else {
+	      auto delta=makeFloat(pr.d_pheader.ts - qd.d_firstquestiontime);
+	      //	      cout<<"Reuse of "<<qi<<", delta t="<<delta<<", count="<<qd.d_qcount<<endl;
+	      if(delta > 2.0) {
+		//		cout<<"Resetting old entry for "<<qi<<", too old"<<endl;
+		qd.d_qcount=0;
+		qd.d_answercount=0;
+		qd.d_firstquestiontime=pr.d_pheader.ts;
+	      }
+	    }
+	    if(qd.d_qcount++)
+              reuses++;
 	  }
 	  else  {  // answer
-	    rcodes[mdp.d_header.rcode]++;
+	    //	    cout<<"Response "<<qi<<endl;
+	    rcodes[header.rcode]++;
 	    answers++;
-	    if(mdp.d_header.rd && !mdp.d_header.ra) {
+	    if(header.rd && !header.ra) {
 	      rdNonRAAnswers++;
 	      rdnonra.insert(pr.getDest());
 	    }
 	  
-	    if(mdp.d_header.ra) {
+	    if(header.ra) {
 	      ComboAddress rem = pr.getDest();
 	      rem.sin4.sin_port=0;
 	      recipients.insert(rem);	  
 	    }
 
 	    QuestionData& qd=statmap[qi];
-
-	    if(!qd.d_qcount)
+	    if(!qd.d_qcount) {
+	      //	      cout<<"Untracked answer: "<<qi<<endl;
 	      untracked++;
+	    }
 
 	    qd.d_answercount++;
 
 	    if(qd.d_qcount) {
 	      uint32_t usecs= (pr.d_pheader.ts.tv_sec - qd.d_firstquestiontime.tv_sec) * 1000000 +  
 		(pr.d_pheader.ts.tv_usec - qd.d_firstquestiontime.tv_usec) ;
-	      //            cout<<"Took: "<<usecs<<"usec\n";
-	      if(usecs<2049000)
-		cumul[usecs]++;
-	      else
-		reallylate++;
```

Comment:
This still prints ```0 responses (0.0000%) older than 2 seconds``` now since we never seem to increment it.

Should that part be removed all together (or add the tracking back in)?